### PR TITLE
[stable/oauth2-proxy] Update to new repo and upstream release

### PR DESCRIPTION
**Pull request by steakunderscore**
#### Is this a new chart

no

#### What this PR does / why we need it:

- oauth2-proxy has it's own github org and with that a new quay container image repo
- v5.1.1 has been released fixing a security issue.

#### Which issue this PR fixes

Minor change so didn't open an issue

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/oauth2-proxy/Chart.yaml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 name: oauth2-proxy
-version: 3.0.0
+version: 3.1.0
```
Comment (steakunderscore) [inline stable/oauth2-proxy/Chart.yaml]: I'm not sure if this is the correct version to bump to. Not sure if this is breaking or not.

**Comment by steakunderscore** [comment]
Unfortunately I don't actually use helm, and don't have the bandwidth to test this. If someone can that'd be great!

**Review by Miouge1 — CHANGES_REQUESTED** [review]
Changing the config file from `/etc/oauth2_proxy/oauth2_proxy.cfg` to `/etc/oauth2-proxy/oauth2-proxy.cfg` is going to be a problem for users using the `config.existingConfig` option.

**Comment by steakunderscore** [comment]
Hmm let me revert changing the file name, that's not really important.

**Comment by steakunderscore** [comment]
Just going through reverting the name change. I'm wondering if it would be a good idea to make this a breaking change, all be it minor, to line up with oauth2-proxy? The reason being is that the name "oauth2_proxy.cfg" is based off the [example config file](), which is now called "oauth2-proxy.cfg.example". I feel like this might catch a few newbies out if they copy paste from the oauth2-proxy docs. WDYT? I'll push up a new commit reverting it, but let me know which way you think would be best to proceed?

**Comment by abdennour** [comment]
Any update on this PR ?
I think it is related to #21198

**Comment by steakunderscore** [comment]
@abdennour Can you test this for us?

**Comment by abdennour** [comment]
@steakunderscore , actually, processing `.Values.extraArgs` is outdated.
Indeed, All args are now started by `-` not `--`.

REFERENCE: [official software doc](https://oauth2-proxy.github.io/oauth2-proxy/auth-configuration)

**Comment by steakunderscore** [comment]
Just noticed this is a duplicate diff of #22309. Perhaps this PR should be closed?

File: stable/oauth2-proxy/Chart.yaml (hunk 1 above)
Comment (desaintmartin) [reply stable/oauth2-proxy/Chart.yaml]: It's acceptable to consider this a feature. This is not breaking (it won't break any existing installation)

**Comment by desaintmartin** [comment]
Your PR is the most complete so far regarding this topic.
so far we still have the following topics to address:
- `.Values.extraArgs` (?)
- config file name with breaking change (if we consider this a breaking change, how to upgrade should be documented in the README). I'm ok to make this a breaking change if @Miouge1 is ok as well.

**Comment by Miouge1** [comment]
> Changing the config file from `/etc/oauth2_proxy/oauth2_proxy.cfg` to `/etc/oauth2-proxy/oauth2-proxy.cfg` is going to be a problem for users using the `config.existingConfig` option.

IMHO and based on the information available so far, the breaking change is not worth it, therefore I would prefer to see this PR as a backwards compatible thing.

**Comment by desaintmartin** [comment]
Should we continue with this PR or should we prefer https://github.com/helm/charts/pull/22309 ?

**Pull request closed without merging**